addons/delivery/models/sale_order.py

```diff
@@ -59,6 +59,10 @@ def set_delivery_line(self, carrier, amount):
         self._remove_delivery_line()
         for order in self:
             order.carrier_id = carrier.id
+            if order.state in ('sale', 'done'):
+                pending_deliverys = order.picking_ids.filtered(
+                    lambda p: p.state not in ('done', 'cancel') and p.location_id == p.picking_type_id.warehouse_id.lot_stock_id)
+                pending_deliverys.carrier_id = carrier.id
```

Comment:
The stock logic of delivery was moved to the module `stock_delivery` recently so this code now belongs in `stock_delivery` (with a new hook method in delivery if really needed) ;).

Reply:
Done